list-agreements.lua

```diff
@@ -1,120 +1,218 @@
+    local del_year, del_year_tick, del_month, del_day
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
+        del_year_tick = agr_year_tick - cur_year_tick
+    end
+    local julian_day = math.floor(del_year_tick / 1200) + 1
+    del_month = math.floor(julian_day / 28)
+    del_day = julian_day % 28
+    return {del_year,del_month,del_day}
 end
 
 function get_guildhall_profession(agr)
-    prof = agr.details[0].data.Location.profession
-    profname = string.lower(df.profession[prof])
+    local prof = agr.details[0].data.Location.profession
+    local profname = string.lower(df.profession[prof])
+    -- *VERY* important code follows
+    if string.find(profname, "man") then
+        profname = string.gsub(profname,"man",string.lower(dfhack.units.getRaceNameById(df.global.ui.race_id)))
+    end
     return profname:gsub("_", " ")
 end
 
-function get_religion_name(agr)
-    religion_id = agr.details[0].data.Location.deity_data.Religion
-    religion_name = dfhack.TranslateName(df.global.world.entities.all[religion_id].name, true)
-    return religion_name
+function get_agr_party_name(agr)
+    --assume party 0 is guild/order, 1 is local government as siteid = playerfortid
+    local party_id = agr.parties[0].entity_ids[0]
+    local party_name = dfhack.TranslateName(df.global.world.entities.all[party_id].name, true)
+    return party_name
 end
 
-function is_satisfied(agr)
-    satisfied = agr.flags.convicted_accepted
-    return satisfied
+function get_deity_name(agr)
+    local religion_id = agr.details[0].data.Location.deity_data.Religion
+    for _,deity_id in ipairs(df.global.world.entities.all[religion_id].relations.deities) do
+        return dfhack.TranslateName(df.global.world.history.figures[deity_id].name,true)
+    end
 end
 
-function is_denied(agr)
-    denied = agr.flags.petition_not_accepted
-    return denied
+
+--return resolution status
+-- 0 no resolution
+-- 1 satisfied
+-- 2 denied
+-- 3 expired
+function is_resolved(agr)
+    local resolution = 0
+    
+    if agr.flags.convicted_accepted then    
+        resolution = 1
+    elseif agr.flags.petition_not_accepted then
+        resolution = 2
+    elseif get_petition_age(agr)[1] ~= 0 then
+        resolution = 3
+    end
+    return resolution
 end
 
-function generate_output_guild(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (denied)")
+--universal handler
+function generate_output(agr,culling,loctype)
```

Comment:
isn't culling already handled in the main script code at the bottom? does it still need to be passed in here?